### ISSUE TITLE
Regression: Reading a parquet file with list of structs results in empty list

### ISSUE THREAD

**Issue by jhorstmann** [pola-rs/polars#12111]
### Checks

- [X] I have checked that this issue has not already been reported.

- [X] I have confirmed this bug exists on the [latest version](https://crates.io/crates/polars) of Polars.

### Reproducible example

Using the [`eventlog.parquet`](https://github.com/pola-rs/polars/files/12845463/eventlog.zip) from #11608

```rust
    let lazy_frame = LazyFrame::scan_parquet("eventlog.parquet", Default::default()).unwrap();
    let mut frame = lazy_frame
        .select([col("case_id"), col("events")])
        .collect().unwrap();

    dbg!(&frame);
```

### Issue description

The output with currently released version is
```
┌──────────┬─────────────────┐
│ case_id  ┆ events          │
│ ---      ┆ ---             │
│ str      ┆ list[struct[2]] │
╞══════════╪═════════════════╡
│ 12345678 ┆ []              │
└──────────┴─────────────────┘
```

### Expected behavior

After the fixes from #11608 and #11619 the output correctly contained the data from the nested struct:

```
┌──────────┬───────────────────────────────────┐
│ case_id  ┆ events                            │
│ ---      ┆ ---                               │
│ str      ┆ list[struct[2]]                   │
╞══════════╪═══════════════════════════════════╡
│ 12345678 ┆ [{"A",1970-01-01 00:00:00.001 UT… │
└──────────┴───────────────────────────────────┘
```

I bisected the regression to commit 2e8354018, the change in `column_idx_to_series` that creates a new `ArrowField` looks related.

### Installed versions

<details>

["dtype-full", "lazy", "parquet", "timezones"]

</details>

**Comment by ritchie46** [comment]
I believe I have fixed this. Can you check against main?

**Comment by jhorstmann** [comment]
Thanks @ritchie46, you are right, I should have checked for changes even though the release was just a few days old.

Fixed by #12048

**Comment by ritchie46** [comment]
I will see if I can patch soon. :+1:
